modules/terraform/output.go

```diff
@@ -49,6 +50,135 @@ func OutputRequiredE(t testing.TestingT, options *Options, key string) (string,
 	return out, nil
 }
 
+func parseListOfMaps(l []interface{}) ([]map[string]interface{}, error) {
+	var result []map[string]interface{}
+
+	for _, v := range l {
+		m, err := parseMap(v.(map[string]interface{}))
+		if err != nil {
+			return nil, err
+		}
+		result = append(result, m)
+	}
+
+	return result, nil
+
+}
+
+// parseMap takes a map of interfaces and parses the types. Is also recursive for nested objects.
+func parseMap(m map[string]interface{}) (map[string]interface{}, error) {
+
+	var result map[string]interface{}
+	result = make(map[string]interface{})
+
+	for k, v := range m {
+		switch v.(type) {
+		case map[string]interface{}:
+			nestedMap, err := parseMap(v.(map[string]interface{}))
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedMap
+		case []interface{}:
+			nestedList, err := parseListOfMaps(v.([]interface{}))
+			if err != nil {
+				return nil, err
+			}
+			result[k] = nestedList
+		case bool:
+			result[k] = v
+		case float64:
+			// Test for int conversion as well. If it can be an int it will be.
```

Comment:
Will all numbers end up with type `float64`? Or will some already be of type `int`? That is, do you need another `case` statement here?

Reply:
The `Unmarshal()` process converts all numbers to floats. So I had to do that `ParseInt` with some error handling to convert ints when I find them.